index/scorch/merge.go

```diff
@@ -82,7 +83,36 @@ OUTER:
 	s.asyncTasks.Done()
 }
 
-func (s *Scorch) planMergeAtSnapshot(ourSnapshot *IndexSnapshot) error {
+func (s *Scorch) parseMergePlannerOptions() *mergeplan.MergePlanOptions {
+	mergePlannerOptions := &mergeplan.DefaultMergePlanOptions
+	scorchOptions := map[string]interface{}{}
+	if v, ok := s.config["scorchOptions"]; ok {
+		if scorchOptions, ok = v.(map[string]interface{}); ok {
+			if v, ok := scorchOptions["maxSegmentsPerTier"].(float64); ok {
```

Comment:
one thought is this approach as a "brittle" feel.  not that merge plan options are changing that often, but if we do add more options, we'll need to remember to update this code here.

one less brittle approach, but perhaps "uglier" is to turn the scorchOptions back into a string, and then just use the standard json parser.  Since this code hardly ever runs (0.0000001% of the time), I can convince myself the laziness tradeoff is worth it

Reply:
(aside: I vaguely recall another near-embarasssing trick (hack), where I needed a "deep copy" of some data structures somewhere, also in a codepath that "hardly ever runs".  Not sure if it was 3 AM or not, but the solution I think I ended up with was to marshal to json and parse it back.  :-) )

Reply:
looks like have to adopt the same approach, just that I do it @3 PM :)